### ISSUE TITLE
top level array type not supported

### ISSUE THREAD

**Issue by OkieOth** [OkieOth/yacg#73]
"IntIntArray": {
      "type": "array",
      "items": {
        "type": "integer"
      }
    }

**Comment by OkieOth** [comment]
Since I am decided against an ArrayType (to avoid breaking changes in the templates) this feature will not be included

**Comment by rbuckland** [comment]
hmm :-) can we re-open this issue ? I was expecting a modelFuncs.isArrayType()
The situation I have is generating plantUML class diagrams 

```
  SomeType:
    description: My Special Type
    type: string
    anyOf:
      - $ref: "../core/base-types.schema.json#/definitions/Whizz"
      - $ref: "../core/base-types.schema.json#/definitions/Bazz"
      - $ref: "../core/base-types.schema.json#/definitions/BangType"
  SomeTypes:
    type: array
    items:
      $ref: "#/definitions/SomeType"
```

and the Diagram I am trying to make is 

<img width="333" alt="image" src="https://user-images.githubusercontent.com/1148383/165912000-188490e7-8442-4f5c-8e7a-d453853bfa60.png">

but alas. Whats the breaking issue n the templates ?

**Comment by OkieOth** [comment]
Hi :) ... the array of array thing was wished by a colleague, so I gave it a try ... but suck :}
The meta model only contains simple types so far (e.g. IntType, ComplexType, ...) and the instances in the parsed models put the used type on the properties.type attrib and annotate the properties in case of arrays about their isArray attrib. 
https://github.com/OkieOth/yacg/blob/master/yacg/model/model.py#L503

For multidimensional arrays I would need a own container/array type in the meta model ... but that would mean to rework all existing templates that are using the isArray approach.

Give me a minute to think about your use-case. IMO it is covered also with the current implementation ... beside the anyOf (I am still afraid to start anyOf implementations ;))

**Comment by OkieOth** [comment]
@rbuckland What about this?
![image](https://user-images.githubusercontent.com/13103714/165936987-17d64589-47d4-4516-87e7-b3f61cfc3a9c.png)

**Comment by OkieOth** [comment]
Version 5.5 will bring this feature

**Comment by rbuckland** [comment]
nice

**Comment by OkieOth** [comment]
If you want you can give the golang_template branch a try. The tests are working so far, but I have not really anyOf or oneOf implemented.
I am going to merge when I have finished a global DictionaryType, will implement it in the next days

**Comment by OkieOth** [comment]
implemented in v5.5.0. In the common puml template, array types and top-level dictionary types are not displayed.
